utility/src/main/java/org/oppia/android/util/profile/ProfileNameValidator.kt

```diff
@@ -0,0 +1,55 @@
+package org.oppia.android.util.profile
+
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Utility to validate that profile names are correctly formatted. */
+@Singleton
+class ProfileNameValidator @Inject constructor() {
+  private val noRepeatedAllowedSymbolsRegex by lazy { Regex("[.'-]{2}") }
+
+  /**
+   * Validates a profile name to ensure that it isn't confusingly formatted
+   * or contains invalid characters.
+   *
+   * @param name name of the profile
+   * @return if the profile name is allowed or not
```

Comment:
Perhaps instead: whether it's a valid, acceptable name?

Reply:
Yep, done :)